lib/ansible/galaxy/role.py

```diff
@@ -44,6 +44,30 @@
 display = Display()
 
 
+class RoleAPI:
+    def __init__(self, preferred_api, api_servers):
+        self._api = preferred_api
+        self.api_servers = api_servers
+
+    @property
+    def api(self):
+        if self._api:
+            return self._api
+
+        for server in self.api_servers:
+            try:
+                if u'v1' in server.available_api_versions:
+                    self._api = server
+                    break
+            except Exception:
```

Comment:
What kind of exception is expected? An attribute error? A type error?

Reply:
I guess any unexpected Exception. This is a copy-paste from https://github.com/ansible/ansible/blob/devel/lib/ansible/cli/galaxy.py#L683-L699.